src/components/ModelDetailsLink/ModelDetailsLink.tsx

```diff
@@ -40,9 +42,11 @@ const ModelDetailsLink = ({
   view,
   ...props
 }: Props) => {
-  const modelDetails = useAppSelector((state) => getModelByUUID(state, uuid));
-  const owner = uuid ? modelDetails?.ownerTag : ownerTag;
-  const model = uuid ? modelDetails?.name : modelName;
+  const { modelName: model, userName: owner } = useModelByUUIDDetails({
+    uuid,
+    ownerTag,
+    modelName,
+  } as ModelByUUIDDetailsProps);
```

Comment:
It could be a little clearer to pass `uuid
      ? { uuid }
      : ({
          ownerTag,
          modelName,
        } as NameProps)`

With possibly also a comment for why the `as NameProps` is necessary (maybe something like "This component's props require either `uuid` or `modelName` to be exist, but this gets lost when the props are destructured."